File: 2099. Find Subsequence of Length K With the Largest Sum.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) 
    {
        int n = nums.size();
        vector<pair<int, int>> vec;
        for(int i = 0; i < n; i++)
            vec.push_back({nums[i], i});

        unordered_set<int>store;
        sort(rbegin(vec), rend(vec));

        for(auto [val, idx] : vec)
        {
            if(!k--) break;
            store.insert(idx);
        }

        vector<int>ans;
        for(int i = 0; i < n; i++)
            if(store.count(i)) ans.push_back(nums[i]);

        return ans;
    }
};

File: 2099. Find Subsequence of Length K With the Largest Sum.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        store = [(nums[i], i) for i in range(n)]
        store.sort()

        index = set()
        for i in range(k):
            index.add(store[n - i - 1][1])

        ans = []
        for i in range(n):
            if len(ans) == k:
                break
            if i in index:
                ans.append(nums[i])
        return ans